Dockerfile

```diff
@@ -17,7 +17,7 @@ WORKDIR /opt/neon-evm/evm_loader
 ARG REVISION
 ENV NEON_REVISION=${REVISION}
 RUN cargo fmt --check && \
-    cargo clippy --release && \
+    cargo clippy --release -- -D warnings && \
```

Comment:
There are 8 crates in this repository. They are listed in: https://github.com/neonlabsorg/neon-evm/blob/0bc93bd9dbc36f49581cf10c51b7498b173ca479/evm_loader/Cargo.toml#L3-L12
Out of the 8 crates, some of them (`lib`, `lib-interface`, `rpc`, `rpc-client`) are not annotated with `#![deny(warnings)]`.

I suggest adding `#![deny(warnings)]` to all crates in this repository, since it makes `cargo clippy` treat warnings as errors both on CI and when run locally.

I also suggest running `cargo clippy --fix` to fix some Clippy warnings introduced both in #400 and in #377. Some warnings like `#[allow(clippy::too_many_arguments)]` need to be fixed by hand.

Basically I would like both `cargo clippy` and `cargo clippy --fix` to run without any reported warnings either locally or on CI after this PR is merged.

I also suggest fixing the Cargo warning: `profiles for the non root package will be ignored, specify profiles at the workspace root`, which was introduced in #412.

All suggestions can be checked on #427. You can cherry-pick them if you agree with my suggestions or just ignore them.